### PR TITLE
[JSON-API] Shutdown on startup if the db connection is invalid

### LINKED ISSUE #10241: JSON API does not fail immediately if connection to DB is broken
As discovered by @akshayshirahatti-da, if you run the JSON API with `createSchema=false` it will happily startup even if it cannot connect to the JSON API. We should try to fail as soon as possible.

### PR DESCRIPTION
**Pull request by realvictorprm**
Should fix #10241

changelog_begin

- [JSON-API] The json api now correctly shutdowns at startup if the provided db connection is invalid in case of `createSchema=false`

changelog_end

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [x] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger-service/http-json/src/main/scala/com/digitalasset/http/Main.scala [hunk 1]
```diff
@@ -123,6 +123,18 @@ object Main {
...
+        Try(dao.isValid(120).unsafeRunSync()).toEither match {
+          case Right(false) =>
+            logger.error("Database connection is not valid.")
+            terminate()
+            System.exit(ErrorCodes.StartupError)
+          case Left(e) =>
+            logger.error("Failed to connect to the database", e)
+            terminate()
+            System.exit(ErrorCodes.StartupError)
+          case _ =>
```
Comment (cocreature) [inline ledger-service/http-json/src/main/scala/com/digitalasset/http/Main.scala:136]:
```suggestion
          case Right(true) =>
```
not a huge fan of wildcard matches when they’re easily avoidable. They tend to bite you when they suddenly end up catching more than you wanted to after refactoring.

**Review by cocreature — APPROVED** [review]
Looks reasonable, thanks! As mentioned on slack, you might be able to test this via the failure tests.

**Comment by realvictorprm** [comment]
ok noticed that I would need to refactor this a bit to be able to test it.

**Comment by realvictorprm** [comment]
And I think it is best to merge this for now, because the refactoring and adding more tests can be done later.

**Pull request merged**